### ISSUE TITLE
Concerns about use of CredentialsContainer

### ISSUE THREAD

**Issue by kenrb** [samuelgoto/WebID#17]
The proposal is currently to add a new credential type to CredentialsContainer (implied by the use of `navigator.credentials.get()`). 

This is going to be confusing for two reasons:

1. There is already a FederatedCredential in CredentialsContainer. Creating a second FederatedCredential with a (presumably) slightly different name but dramatically different behaviour is clearly not ideal.
2. This doesn't actually store a credential, which is what CredentialsContainer currently does, viz. this would be the only type with `get()` calls but no corresponding `store()` calls. This is stretching the existing abstraction pretty far.

AFAICT the reason to use CredentialsContainer is that WebID is a related concept (obtaining information for the purpose of user authentication) but if that is the only reason then I think the above reasons are a strong case for using a new construct.

**Comment by samuelgoto** [comment]
Yep, I think those are fair points ... I can certainly empathize with some of those points. 

I used `navigator.credentials` as a starting point per @mikewest's recommendation. I agree that it is not perfect, but seems like a reasonable starting point until we learn more about the construct.

I'm using `navigator.credentials` for WebOTP and we use it for WebAuthn too, so seemed like at least a reasonable approximation and a reasonable home for this.

Do you have a suggestion that you think works best then `navigator.credentials`? personas used `navigator.id` which I think seemed nice. other suggestions?

**Comment by mikewest** [comment]
1. My goal was to extend `FederatedCredential` in this direction (see https://www.w3.org/TR/credential-management-1/#teh-futur). Given that there's practically zero usage of that API as it exists today, I think it's quite reasonable to simply extend or replace it. I was assuming you'd just add whatever information to that object. Is that not a good idea, given your current model?

2. Some kind of `store()` call seems helpful on pages that support multiple IDPs, as it addresses the "Hrm. Did I use Twitter or GitHub to log in here?" problem.

I'm not dogmatic about y'all using `navigator.credentials`. If something else works better for you, great! That said, I think the confusion argument cuts the other way. Given that `navigator.credentials` does exist in all browsers, wouldn't it be more confusing to separate out this kind of credential as special and requiring a completely different API?

**Comment by kenrb** [comment]
That's interesting, I hadn't thought about trying to extend `FederatedCredential` while keeping the old behaviour intact. Replacing it outright would break existing use (such as it is), right?

If we can version the interface then that might make the most sense.

**Comment by samuelgoto** [comment]
Seems like we found some converge, so I added a section [navigator.id](https://github.com/samuelgoto/WebID#navigator-id) on the [Alternatives Considered](https://github.com/samuelgoto/WebID#alternatives-considered) to capture this discussion and where we stand at the moment. I think it is worth noting that this is an open question, as it has come up multiple times. 

Happy to re-open this and re-evaluate, but wanted to use mike's guidance as a starting point and go from there once we know more.
